src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentStaticPageDirective.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable disable
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Language.Components;
+
+internal class ComponentStaticPageDirective
+{
+    public static readonly DirectiveDescriptor Directive = DirectiveDescriptor.CreateDirective(
+        "staticpage",
+        DirectiveKind.SingleLine,
+        builder =>
+        {
+            builder.Usage = DirectiveUsage.FileScopedSinglyOccurring;
+            builder.Description = ComponentResources.StaticPageDirective_Description;
+        });
```

Comment:
You might wonder if this should also take a bool flag, so you can set `@staticpage true` or `@staticpage false`.

I think it should *not* take such a flag because:

 * Setting `@staticpage false` might mislead people into thinking they are making the page "not static" somehow but what it really would mean is "don't apply the 'static' override"
 * Longer term, we might want to accept some parameter here to control this feature in more detail, but it likely wouldn't be a bool

So I propose to keep this simple and only support `@staticpage`. Once it's set for a page, you can't unset it - there's no requirement to allow that now, and if we want to add that in the future, it probably would be something like an enum rather than a bool.